.github/workflows/tests.yml

```diff
@@ -0,0 +1,49 @@
+name: Tests
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        elixir: ["1.13", "1.14"]
+        otp: ["23", "24", "25"]
+        include:
+          - nsq_download: nsq-0.3.8.linux-amd64.go1.6.2
+            nsq_version: v0.3.8
+            worker_id: worker-id
+          - nsq_download: nsq-1.0.0-compat.linux-amd64.go1.8
+            nsq_version: v1.0.0-compat
+            worker_id: node-id
+    env:
+      MIX_ENV: test
+      NSQ_DOWNLOAD: ${{ matrix.nsq_download }}
+      NSQ_VERSION: ${{ matrix.nsq_version }}
+      WORKER_ID: ${{ matrix.worker_id }}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp }}
+          elixir-version: ${{ matrix.elixir }}
+      - run: wget https://github.com/nsqio/nsq/releases/download/$NSQ_VERSION/$NSQ_DOWNLOAD.tar.gz
+      - run: tar xvzf $NSQ_DOWNLOAD.tar.gz
+      - run: mix deps.get
+      - run: mix compile
+      - run: mix run --no-halt -e "NSQ.Test.AuthServer.start(6790)" &
+      - run: $NSQ_DOWNLOAD/bin/nsqd --tcp-address=127.0.0.1:6750 --http-address=127.0.0.1:6751 --$WORKER_ID=512 --data-path=test/data_paths/512 --broadcast-address=127.0.0.1 --lookupd-tcp-address=127.0.0.1:6770 --lookupd-tcp-address=127.0.0.1:6780 &
+      - run: $NSQ_DOWNLOAD/bin/nsqd --tcp-address=127.0.0.1:6760 --http-address=127.0.0.1:6761 --$WORKER_ID=513 --data-path=test/data_paths/513 --broadcast-address=127.0.0.1 --lookupd-tcp-address=127.0.0.1:6770 --lookupd-tcp-address=127.0.0.1:6780 &
+      - run: $NSQ_DOWNLOAD/bin/nsqd --tcp-address=127.0.0.1:6765 --http-address=127.0.0.1:6766 --https-address=127.0.0.1:6767 --$WORKER_ID=514 --data-path=test/data_paths/514 --broadcast-address=127.0.0.1 --lookupd-tcp-address=127.0.0.1:6770 --lookupd-tcp-address=127.0.0.1:6780 --tls-required=false --tls-root-ca-file=test/ssl_keys/elixirNsq.pem --tls-key=test/ssl_keys/elixir_nsq.key --tls-cert=test/ssl_keys/elixir_nsq.crt --auth-http-address=127.0.0.1:6790 &
+      - run: $NSQ_DOWNLOAD/bin/nsqlookupd --tcp-address=127.0.0.1:6770 --http-address=127.0.0.1:6771 --broadcast-address=127.0.0.1 &
+      - run: $NSQ_DOWNLOAD/bin/nsqlookupd --tcp-address=127.0.0.1:6780 --http-address=127.0.0.1:6781 --broadcast-address=127.0.0.1 &
```

Comment:
You can define a service like this: https://github.com/wistia/bottler/blob/master/.github/workflows/elixir.yml#L17-L32

I'd recommend doing something like that with nsq and using a prebuilt docker image like the one we use in grand-central. https://github.com/wistia/grand-central/blob/master/docker-compose.yml#L77-L103

Reply:
I just copied what was in travis CI but I can clean it up to be more modern.

Reply:
@blatyo I'm going to punt on this since I'm not sure how to make the services talk to each other since github actions doesn't let you specify the --network option. The services need to be able to talk to nsqlookup. Let me know if you have any suggestions.

Reply:
All the services are on the same network in GHA. Your service name is the DNS address you can use to reference it and you specify the port that you want to expose.

It's not a big deal though.